### PR TITLE
Support for DRTIO 100MHz

### PR DESCRIPTION
**Pull request by Spaqin**
# ARTIQ Pull Request

## Description of Changes

This pull request adds support for 100MHz DRTIO clock for KC705 and Kasli. The PLL and VCO frequency parameters have been checked with the datasheets for these boards and it's well within specs.

For Kasli, frequency is chosen within the JSON file ("rtio_frequency" key); for KC705 it's an additional argument for building.

Additionally, this PR includes changes in the siphaser, meaning it must be merged before the corresponding PR in artiq-zynq repo.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :sparkles: New feature |

### All Pull Requests

- [x] Use correct spelling and grammar.
- [x] Update [RELEASE_NOTES.rst](../RELEASE_NOTES.rst) if there are noteworthy changes, especially if there are changes to existing APIs.
- [x] Close/update issues.

### Code Changes

- [x] Run `flake8` to check code style (follow PEP-8 style). `flake8` has issues with parsing Migen/gateware code, ignore as necessary.
- [x] Test your changes or have someone test them. Mention what was tested and how.
- [x] Add and check docstrings and comments

### Git Logistics

- [x] Split your contribution into logically separate changes (`git rebase --interactive`). Merge/squash/fixup commits that just fix or amend previous commits. Remove unintended changes & cleanup. See [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
- [x] Write short & meaningful commit messages. Review each commit for messages (`git show`). Format:
  ```
  topic: description. < 50 characters total.
  
  Longer description. < 70 characters per line
  ```

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

File: artiq/firmware/satman/main.rs [hunk 1]
```diff
@@ -447,6 +447,19 @@ const SI5324_SETTINGS: si5324::FrequencySettings
...
+const SI5324_SETTINGS: si5324::FrequencySettings
+    = si5324::FrequencySettings {
+    n1_hs  : 5,
+    nc1_ls : 10,
+    n2_hs  : 10,
+    n2_ls  : 250,
+    n31    : 50,
+    n32    : 50,
+    bwsel  : 4,
+    crystal_ref: true
```
Comment (sbourdeauducq) [inline artiq/firmware/satman/main.rs:460]: How did you determine these?

Comment (Spaqin) [reply artiq/firmware/satman/main.rs:460]: I used DSPLL sim program from Silicon Laboratories (which miraculously worked under Wine, more or less) - put in the input clock freq (100MHz from siphaser), desired output, and it calculated these values for me. I verified that they're mapped by the driver to proper register values as well.

Comment (sbourdeauducq) [reply artiq/firmware/satman/main.rs:460]: OK great

**Comment by jbqubit** [comment]
> This pull request adds support for 100MHz DRTIO clock for KC705 and Kasli. 

Please test changes like this on hardware before merging. Checking manual and simulation is half way...

**Comment by jbqubit** [comment]
Support for 100 MHz /is/ quite helpful.

**Comment by sbourdeauducq** [comment]
> Please test changes like this on hardware before merging.

@Spaqin has tested this already, both as sat and master, between zc706 and kc705. This was a request from NIST and those are the boards they are using.

**Pull request merged**